docs/releases/nemotronmathproofs/index.md

```diff
@@ -0,0 +1,390 @@
+
+The prover pipeline generates proofs for formalized statements with iterative error correction.
+
+=== "CLI"
+
+    ```bash
+    ns generate \
+        --cluster=slurm \
+        --generation_module=nemo_skills.inference.prover \
+        --model=Goedel-LM/Goedel-Prover-V2-32B \
+        --server_type=vllm \
+        --server_gpus=8 \
+        --server_args="--max-model-len 40960" \
+        --input_file=/workspace/data/formal_statements.jsonl \  # Output from autoformalization step
+        --output_dir=/workspace/data/proofs_output \
+        --with_sandbox \
+        --num_random_seeds=1 \
+        --dependent_jobs=2 \
+        ++prompt_config=lean4/goedel-prover-v2 \
+        ++inference.tokens_to_generate=38912 \
+        ++inference.temperature=1.0 \
+        ++inference.top_p=0.95 \
+        ++refinement=True \
+        ++refinement_max_turns=8 \
+        ++remove_cot=True \
+        ++n_pass=1 \
+        ++refinement_prompt_config=lean4/goedel-prover-v2-refinement \
+        ++delete_wrong_turns=True \
+        ++max_concurrent_requests=512
+    ```
+
+=== "Python"
+
+    ```python
+    from nemo_skills.pipeline.cli import generate, wrap_arguments
+
+    generate(
+        ctx=wrap_arguments(
+            "++inference.tokens_to_generate=38912 "
+            "++inference.temperature=1.0 "
+            "++inference.top_p=0.95 "
+            "++prompt_config=lean4/goedel-prover-v2 "
+            "++refinement=True "
+            "++refinement_max_turns=8 "
+            "++remove_cot=True "
+            "++n_pass=1 "
+            "++refinement_prompt_config=lean4/goedel-prover-v2-refinement "
+            "++delete_wrong_turns=True "
+            "++max_concurrent_requests=512 "
+        ),
+        generation_module="nemo_skills.inference.prover",
+        cluster="slurm",
+        input_file="/workspace/data/formal_statements.jsonl",  # Output from autoformalization step
+        output_dir="/workspace/data/proofs_output",
+        model="Goedel-LM/Goedel-Prover-V2-32B",
+        server_type="vllm",
+        server_gpus=8,
+        server_args="--max-model-len 40960",
+        num_random_seeds=1,
+        dependent_jobs=2,
+        with_sandbox=True,
+    )
+    ```
+
+The proving strategy includes:
+
+* **Chain-of-thought removal**: Strips reasoning, keeping only formal proof code
+* **Wrong turn deletion**: Discards failed attempts to prevent context pollution
+* **Structured error feedback**: Compiler errors annotated with `<error></error>` tags
+* **Pass@N with refinement**: Multiple independent attempts, each with iterative refinement
+
+### Model Training
+
+To fine-tune a model on the Nemotron-Math-Proofs dataset:
+
+=== "CLI"
+
+    ```bash
+    ns sft_nemo_rl \
+        --cluster=slurm \
+        --expname=qwen3-8b-lean-sft \
+        --output_dir=/workspace/training/qwen3-8b-lean-sft \
+        --hf_model=Qwen/Qwen3-8B \
+        --training_data=/workspace/data/sft_data.jsonl \  # Processed output from theorem proving step
```

Comment:
We should ideally add instructions with a simple command for how to construct this data using the release HF dataset. Probably just a few lines to download / lightly postprocess. See other releases for examples

Reply:
ideally also add a step to prepare the output of theorem proving into this format, but it's less important than how to use existing dataset